pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/FilterOperatorUtils.java

```diff
@@ -58,13 +57,21 @@ public static BaseFilterOperator getLeafFilterOperator(PredicateEvaluator predic
   }
 
   /**
-   * Re-order filter operators based on the their cost. Put the ons with inverted index first so we can process less
+   * Returns whether to use scan re-order optimization (ON by default) based on the given debug options.
+   */
+  public static boolean enableScanReorderOptimization(@Nullable Map<String, String> debugOptions) {
```

Comment:
I understand why you pass the whole debugOptions instead of just a boolean value. I keep the debugOptions in the operator, but added a method to extract the scanReorderOptimization flag from the debugOptions. For other options, we can add other methods that take debugOptions as input and extract the option from it.